server/channels/app/server.go

```diff
@@ -1152,7 +1152,31 @@ func (a *App) OriginChecker() func(*http.Request) bool {
 
 		return utils.OriginChecker(allowed)
 	}
-	return nil
+
+	// Overriding the default origin checker
+	return func(r *http.Request) bool {
+		origin := r.Header["Origin"]
+		if len(origin) == 0 {
+			return true
+		}
+		if origin[0] == "null" {
+			return false
+		}
```

Comment:
What is this case covering exactly? It's not super clear from this side of things.

Reply:
It got lost in the previous discussion https://github.com/mattermost/mattermost-server/pull/23489#discussion_r1204051836

Reply:
For additional context, MDN does a good job of explaining null origins: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description

Reply:
Thanks both :+1: